shell_integration/windows/OCUtil/RemotePathChecker.cpp

```diff
@@ -182,9 +187,8 @@ RemotePathChecker::~RemotePathChecker()
     CloseHandle(_newQueries);
 }
 
-vector<wstring> RemotePathChecker::WatchedDirectories()
+std::shared_ptr<const std::vector<std::wstring>> RemotePathChecker::WatchedDirectories() const
 {
-    std::unique_lock<std::mutex> lock(_mutex);
```

Comment:
Still need the lock!

The refcount is atomic, meaning that different instance of the shared_ptr pointing to the same object can be used in different thread.  But the same shared_ptr instance cannot, it is not thread safe.